indra/llmessage/llpacketbuffer.cpp

```diff
@@ -67,10 +67,25 @@ LLPacketBuffer::~LLPacketBuffer ()
 
 ///////////////////////////////////////////////////////////
 
-void LLPacketBuffer::init (S32 hSocket)
+void LLPacketBuffer::init(S32 hSocket)
 {
     mSize = receive_packet(hSocket, mData);
     mHost = ::get_sender();
     mReceivingIF = ::get_receiving_interface();
 }
 
+void LLPacketBuffer::init(char* buffer, S32 data_size, const LLHost& host)
+{
+    if (data_size > NET_BUFFER_SIZE)
+    {
+        LL_ERRS() << "Initializing packet with size=" << data_size << " > " << NET_BUFFER_SIZE << LL_ENDL;
+    }
+    else
+    {
+        memcpy(mData, buffer, data_size);
+        mSize = data_size;
+        mHost = host;
+        mReceivingIF = ::get_receiving_interface();
+    }
+}
```

Comment:
I'm adding this `LLPacketBuffer::init()` method to allow the `LLPacketRing` to recycle existing instances for new packets.

Reply:
May be add that comment before the memcpy?